Plugins: Only offer email support for monthly plans

epeicher opened the pull request:
#### Changes proposed in this Pull Request

* The plugin details page will show `Unlimited Email Support` for monthly
plans and `Live chat support 24x7` for Annual plans.

|Before | After|
|-------|------|
|![image](https://user-images.githubusercontent.com/3519124/154542415-1f6df76d-5ff6-4b37-99cd-0a1915f9f555.png)|![image](https://user-images.githubusercontent.com/3519124/154541497-d17099e6-9092-4ff6-87d4-de6a173f27d4.png)|
|![image](https://user-images.githubusercontent.com/3519124/154542016-84d37b4c-c4c5-4642-b619-c0b99079f3ec.png)|![image](https://user-images.githubusercontent.com/3519124/154541254-bcc4e304-a214-4713-b45c-57116da11d2c.png)|

#### Testing instructions

* In a Business or eCommerce plan
* Go to plugins and add a Premium plugin
* Select Monthly period for the plugin
  * There should be a message in the Plugin details section (see screenshots above) that reads: `Unlimited Email Support`  
* Select Annually period for the plugin
  * There should be a message in the Plugin details section (see screenshots above) that reads: `Live chat support 24x7`


Fixes #59753

epeicher committed:
Plugins: Only offer email support for monthly plans

The plugin details page will show `Unlimited Email Support` for monthly
plans and `Live chat support 24x7` for Annual plans.

This change is applied to both sections, the top right `Purchase and activate`
and the Plugin details section in the bottom right.

Fixes #59753

epeicher committed:
Plugins: Remove a warning in development console

Add a key prop to a list of React elements as suggested by a React
warning to avoid errors during development and to prevent potential bugs.

WBerredo commented on client/my-sites/plugins/plugin-details-sidebar/index.jsx:
```diff
@@ -91,7 +95,11 @@ const PluginDetailsSidebar = ( {
-				description={ translate( 'Live chat support 24x7.' ) }
+				description={
+					isAnnualPeriod
+						? translate( 'Live chat support 24x7.' )
+						: translate( 'Unlimited Email Support' )
```
Nitpicking: I'm not sure about this `.` in the end of `'Live chat support 24x7.'`. 
It is not present on the other place and none of the `'Unlimited Email Support'` has this trailing dot.

I would go for removing it.

epeicher replied:
Good catch. That message was like that before my change, but I agree it is not consistent with other appearances of the same message, so I will go ahead and remove the `.`

WBerredo reviewed (approved):
LGTM! Everything working as expected.

epeicher committed:
Remove a trailing dot in message for consistency

It is not present in other appearances of the same message

cpapazoglou commented on client/my-sites/plugins/plugin-details-sidebar-usp/index.jsx:
```diff
@@ -69,14 +70,14 @@ const PluginDetailsSidebarUSP = ( { id, icon, title, description, links, first }
-				links.map( ( link ) => {
+				links.map( ( link, idx ) => {
 					return (
-						<>
+						<Fragment key={ idx }>
 							<ExternalLink icon href={ link.href }>
 								{ link.label }
 							</ExternalLink>
 							<br />
-						</>
+						</Fragment>
```
Thanks for fixing this.

cpapazoglou commented on client/my-sites/plugins/plugin-details-sidebar/index.jsx:
```diff
@@ -14,9 +15,12 @@ const PluginDetailsSidebar = ( {
 		demo_url = null,
 		documentation_url = null,
 	},
+	billingPeriod,
```
Instead of passing the current `billingPeriod` as a prop, we have a state selector we can use

https://github.com/Automattic/wp-calypso/blob/0753f8792dc14abc16d52b99781b7ddb15bd9205/client/my-sites/plugins/plugins-browser/index.jsx#L138

cpapazoglou reviewed (changes requested):
Looks and works great, thanks @epeicher!

I have added a comment regarding the usage of `const billingPeriod = useSelector( getBillingInterval );`

epeicher committed:
Use selector instead of passing a prop to read billingPeriod

The billingPeriod can be read from the redux state instead of being
passed from parent to child as suggested by a comment.

cpapazoglou reviewed (approved):
Thanks for addressing my feedback.

LGTM!
